### PR TITLE
Start of unit test generation

### PR DESCRIPTION
**Pull request by chrisdunelm**
Only generates sync request-method generation at the moment; and is not yet well-tested.

### CODE REVIEW

File: Google.Api.Generator.Tests/ProtoTest.cs [hunk 1]
```diff
@@ -94,11 +94,9 @@ private void ProtoTestSingle(string testProtoName, bool ignoreCsProj = false, bo
             // Verify each output file.
             foreach (var file in files)
             {
-                if (ignoreCsProj && file.RelativePath.EndsWith(".csproj"))
-                {
-                    continue;
-                }
-                if (ignoreSnippets && file.RelativePath.Contains($".Snippets{Path.DirectorySeparatorChar}"))
+                if (ignoreCsProj && file.RelativePath.EndsWith(".csproj") ||
```
Comment (jskeet) [inline Google.Api.Generator.Tests/ProtoTest.cs]: I honestly can't remember precedence rules for `&&` and `||`, but I think the code would at least be clearer with this being explicit:

```csharp
if ((ignoreCsProj && file.RelativePath.EndsWith(".csproj")) ||
    (... && ...) ||
    (... && ...))
```

Comment (chrisdunelm) [reply Google.Api.Generator.Tests/ProtoTest.cs]: Oops, yes, extra parentheses added.

**Comment by chrisdunelm** [comment]
PTAL

File: Google.Api.Generator.Tests/ProtoTest.cs (hunk 1 above)
Comment (chrisdunelm) [reply Google.Api.Generator.Tests/ProtoTest.cs]: And something you can't remember about C#! Incredible!!

**Comment by jskeet** [comment]
(Was interrupted when reviewing, hence single comment. More later.)

**Pull request merged**